virttest/compat_52lts.py

```diff
@@ -0,0 +1,42 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2018
+# Author: Lukas Doktor <ldoktor@redhat.com>
+
+"""
+This module contains helpers that allows running Avocado-vt with Avocado
+master as well as with 52.x LTS release.
+"""
+
+
+def results_stdout_52lts(result):
+    """
+    Get decoded stdout text in 52.x LTS backward compatible way
+
+    :param result: result object
+    """
+    if hasattr(result, "stdout_text"):
+        return result.stdout_text
+    else:   # 52lts stores string
+        return result.stdout
+
+
+def results_stderr_52lts(result):
```

Comment:
Better we add one more method to make whole result compatible  with 52.x since in some method cmdResult object is returned ,rather than stdout or stderr.
```
def results_52lts(result):
       if hasattr(result, "stdout_text"):
               result.stdout = result.stdout_text
               result.stderr = result.stderr_text
      return result
```

Reply:
Well modifying the results would be another approach but it'd have to be another object (on newer Avocados it'd result in `result.stdout_text` calling decode on decoded string) and I was afraid it'd be less explicit. For that reason I think we better use the separate `stdout/stderr` wrapper and later we simply replace them with `result.stdout_text` methods.

(the need to call this explicitly forces people to think whether text output is necessary. You can see it in this commit, not all places require text output and bytes are enough so eventually when we deprecate 52 it should be easier to find the right spots. There is a great difference between `stdout` and `stdout_text` on py3 and we'll have to deal with it further on)

Reply:
I mean besides results_stdout_52 and s results_stderr_52 two approach, we may take this opportunity to expose one more approach results_52lts for compatible between py2 and  py3.
For example, in virsh.py under avocado-vt, we hope that the return cmdResult from command(cmd, **dargs) should be text whether it is py2 or py3. Therefore, this approach can provide this convenience.

Reply:
But that is what I'm afraid of. Providing such method would lead to usage (in tests) where people would not be forced to distinguish between text and non-text access. It'd be simpler now but way harder once we really move to py3 and non-52.lts compat. (with this patch people have to explicitly ask for text version, with your proposal they get it "by default" without thinking but we need to start thinking about it and use the encoding only when needed).

Anyway such change can be proposed in a separate PR, but I'd encourage you not to do so as it'll only bring problems later when we move to py3.